src/stream.rs

```diff
@@ -15,19 +16,19 @@ pub enum MaybeHttpsStream<T> {
     /// A stream over plain text.
     Http(T),
     /// A stream protected with TLS.
-    Https(TlsStream<T>),
+    Https(TokioIo<TlsStream<TokioIo<T>>>),
 }
 
-impl<T: AsyncRead + AsyncWrite + Connection + Unpin> Connection for MaybeHttpsStream<T> {
+impl<T: Read + Write + Connection + Unpin> Connection for MaybeHttpsStream<T> {
```

Comment:
Nit: I'd prefer to use `rt::Read + rt::Write` since the trait names are also used in `std` and I'd like to clearly disambiguate from those.

(Same for the other uses below.)

Reply:
Done and squashed.